tests/parser/UDQTests.cpp

```diff
@@ -2345,13 +2510,25 @@ UDQ
     auto undefined_value =  udq.params().undefinedValue();
     UDQState udq_state(undefined_value);
     SummaryState st(TimeService::now());
+
+    const auto required_keys = [&udq]()
     {
-        std::unordered_set<std::string> required_keys;
-        udq.required_summary(required_keys);
-        BOOST_CHECK(required_keys.empty());
-    }
+        auto keys = std::unordered_set<std::string>{};
+        udq.required_summary(keys);
+
+        auto required = std::vector<std::string>{ keys.begin(), keys.end() };
+        std::sort(required.begin(), required.end());
 
-    udq.eval(0, schedule, {}, st, udq_state);
+        return required;
+    }();
+
+    const auto expected_keys = std::vector<std::string> {};
+
+    BOOST_CHECK_EQUAL_COLLECTIONS(required_keys.begin(), required_keys.end(),
```

Comment:
Strange, that this compiles. Isn't required_keys a lambda now?

Reply:
> Isn't `required_keys` a lambda now?

It's subtle, and possibly a bit too cute, but please note the [final two parentheses here](https://github.com/OPM/opm-common/blob/381ab825b1b4f01ebaec77655fc8c57edc24ea5c/tests/parser/UDQTests.cpp#L2268).  This makes the overall structure of the statement be
```C++
const auto required_keys = [&udq]() { return stuff(); }();
```
and makes the lambda into an [immediately invoked function expression](https://en.wikipedia.org/wiki/Immediately_invoked_function_expression) instead.  This means we care only about the result of calling the lambda rather than the lambda itself.  One benefit of this is that we can mark `required_keys` as `const` whereas the full construction requires mutability of the vector.

The number one downside to this construction is that those two parentheses are often hard to spot when reading the code.

Reply:
They indeed are hard to overlook.